javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -135,14 +133,11 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     @Override
     public T get() {
-        // using a local var speeds up the double-check idiom by 25%, see Effective Java, Item 71
-        Supplier<? extends T> tmp = supplier;
-        if (tmp != null) {
+        if ((value == null) && (supplier != null)) {
```

Comment:
@danieldietrich, pushed a new version - though I'm not sure about speed or correctness anymore :p
FYI, `get` is small enough to be inlinable now , i.e. `javaslang.Lazy::get (19 bytes) inline (hot)`